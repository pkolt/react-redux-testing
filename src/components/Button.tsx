import React from 'react';

type ButtonProps = React.PropsWithChildren<{
    modifier?: string;
}>

const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
    const { modifier, children } = props;
    return (
        <button type="submit" className={`btn ${modifier ?? ''}`}>{children}</button>
    );
}

export default Button;
