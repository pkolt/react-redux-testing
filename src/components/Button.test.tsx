import React from 'react';
import Button from './Button';
import { shallow } from 'enzyme';

describe('<Button />', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
  });
});
