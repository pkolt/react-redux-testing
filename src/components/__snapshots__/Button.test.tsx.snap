// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders without crashing 1`] = `ShallowWrapper {}`;
